modules/library/src/main/scala/zio/elasticsearch/query/Queries.scala

```diff
@@ -584,6 +585,25 @@ private[elasticsearch] final case class MatchPhrase[S](field: String, value: Str
     )
 }
 
+sealed trait MultiMatchQuery[S] extends ElasticQuery[S] with HasFields[MultiMatchQuery[S]]
+
+private[elasticsearch] final case class MultiMatch[S](fields: Option[Chunk[String]], value: String)
+    extends MultiMatchQuery[S] { self =>
+  def fields(fields: String*): MultiMatchQuery[S] =
+    self.copy(fields = Some(Chunk.fromIterable(fields)))
+
+  def fields[S1: Schema](fields: Field[S1, String]*): MultiMatchQuery[S] =
+    self.copy(fields = Some(Chunk.fromIterable(fields.map(_.toString))))
```

Comment:
Also, some sort of restrictions should be discussed, as (I think) we can not have both type-safe and raw fields. @arnoldlacko what do you think?

Reply:
I think given that we use `_.toString` here, the resulting query is the same regardless of being constructed using Field or raw strings, so it does not need to be an issue if they are mixed.
What I wonder is why we have a `S1` type here with no defined relation to `S`, and why it does not affect the type of the resulting query.
@mvelimir any thoughts?

Reply:
We'll definitely need `S1` to be a subtype of `S`, to narrow the type which will always start from `Any` based on the constructor method. Also, I think mixing both type-safe fields and plain strings should be allowed, since strings in this case can be used as wildcards, and having type-safe fields will still be useful in inferring the type of the document. Also, the type-safe method should definitely contain at least one field as to not clash with the strings method when the argument list is empty. With all this in mind, I think the method signature should be:
```scala
def fields[S1 <: S: Schema](field: Field[S1, String], fields: Field[S1, String]*): MultiMatchQuery[S1]
```